include/nbl/video/IGPUObjectFromAssetConverter.h

```diff
@@ -327,17 +330,115 @@ auto IGPUObjectFromAssetConverter::create(const asset::ICPUBuffer** const _begin
 
     return res;
 }
+namespace impl
+{
+template<typename MapIterator>
+struct CustomBoneNameIterator
+{
+        inline CustomBoneNameIterator(const MapIterator& it) : m_it(it) {}
+        inline CustomBoneNameIterator(MapIterator&& it) : m_it(std::move(it)) {}
+
+        inline bool operator!=(const CustomBoneNameIterator<MapIterator>& other) const
+        {
+            return m_it!=other.m_it;
+        }
+
+        inline CustomBoneNameIterator<MapIterator>& operator++()
+        {
+            ++m_it;
+            return *this;
+        }
+        inline CustomBoneNameIterator<MapIterator> operator++(int)
+        {
+            return m_it++;
+        }
+
+        inline CustomBoneNameIterator<MapIterator> operator-(const CustomBoneNameIterator<MapIterator>& other) const
```

Comment:
i think return type should be ptrdiff_t or something, not another iterator for sure
actually im almost sure that only random access iterators have `operator-` defined

Reply:
is a map iterator a random access iterator?

Reply:
nope